backend/rust/src/main.rs

```diff
@@ -0,0 +1,88 @@
+use backend::backend_server::{Backend, BackendServer};
+use backend::{HealthMessage, PredictOptions,Reply,ModelOptions, EmbeddingResult, GenerateImageRequest,TranscriptRequest,TranscriptResult,TtsRequest,TokenizationResponse};
+use tonic::{Request, Response, Status};
+use tokio_stream::{wrappers::ReceiverStream};
+
+use tonic::transport::Server;
+
+pub mod backend{
+    tonic::include_proto!("backend");
+}
+
+
+#[derive(Debug)]
+struct BackendService;
+
+#[tonic::async_trait]
+impl Backend for BackendService{
+
+    // Result in proto/backend.rs is conflict with std::result::Result
+    // So we need to use use the fully qualified name of the Result type in the protobuf file
+
+    async fn health(&self, request: Request<HealthMessage>) -> Result<Response<Reply>,Status> {
+        
+        //TODO: Maybe we can move this to a logger
```

Comment:
Since it is already using tokio, `tracing` and `tracing-subscriber` feel a natural choice.

Reply:
Done